crates/ide/src/goto_definition.rs

```diff
@@ -29,7 +35,11 @@ pub(crate) fn goto_definition(
     let file = sema.parse(position.file_id).syntax().clone();
     let original_token = pick_best(file.token_at_offset(position.offset))?;
     let token = sema.descend_into_macros(original_token.clone());
-    let parent = token.parent();
+    let parent = if let Some(comment) = ast::Comment::cast(token.clone()) {
+        resolve_intra_doc_links(position, comment, file, token.clone())
+    } else {
+        token.parent()
+    };
```

Comment:
```suggestion
    if let Some(comment) = ast::Comment::cast(token.clone()) {
        return resolve_intra_doc_links(position, comment, file, token.clone())
    }
    let parent =  token.parent();
```

doc links resolving should be a completely separate code path I think.